regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java

```diff
@@ -41,10 +42,17 @@ public EmptyAlternativeFinder(RegexIssueReporter.ElementIssue regexElementIssueR
   @Override
   public void visitDisjunction(DisjunctionTree tree) {
     List<RegexTree> alternatives = tree.getAlternatives();
-    alternatives.stream().limit((alternatives.size() - 1))
-        .filter(alternative -> alternative.is(RegexTree.Kind.SEQUENCE))
-        .filter(alternative -> ((SequenceTree) alternative).getItems().isEmpty())
-        .forEach(alternative -> regexElementIssueReporter.report(alternative, MESSAGE, null, Collections.emptyList()));
+    for (int i = 0; i < alternatives.size() - 1; i++) {
+      if (isEmptyAlternative(alternatives.get(i))) {
+        SourceCharacter orOperator = tree.getOrOperators().get(i);
```

Comment:
If I understand correctly, here you highlight the orOperator (the '|' character) to the right of the empty alternative?

Reply:
We should be careful to report issues on trees that have no range. This verification inside the check verifier will show us at an early stage when we try to highlight a tree without range.